ehcache-impl/src/main/java/org/ehcache/config/builders/UserManagedCacheBuilder.java

```diff
@@ -166,189 +157,183 @@ private UserManagedCacheBuilder(UserManagedCacheBuilder<K, V, T> toCopy) {
     this.sizeOfUnit = toCopy.sizeOfUnit;
   }
 
+  @SuppressWarnings("try")
   T build(ServiceLocator.DependencySet serviceLocatorBuilder) throws IllegalStateException {
 
-    validateListenerConfig();
+    id = Optional.ofNullable(id).orElse("UserManagedCache - " + instanceId.incrementAndGet());
```

Comment:
You can't set the `id` if it hasn't been explicitly set by the user. That will mess with the logic further down for requiring an explicit `id` be set for persistent caches. Something like this should work:
```java
String alias = id == null ? "UserManaged" + instanceId.getAndIncrement() : id;
```